File: src/main/java/Items/FruitType.java
package Items;

import java.util.Random;

import javafx.scene.image.Image;


/**
 * フルーツの種類を表す列挙型。
 * 種類ごとにスコアと色（仮描画用）を持つ。
 */
public enum FruitType {
    CHERRY(100, "/picture/sakuranbo.png"),         // サクランボ
    STRAWBERRY(300, "/picture/ichigo.png"), // イチゴ
    ORANGE(500, "/picture/orange.png"),         // オレンジ
    APPLE(700, "/picture/ringo.png"),           // リンゴ
    GRAPE(1000, "/picture/budo.png");          // ブドウ

    private final int score;
    private final Image image;

    FruitType(int score, String imagePath) {
        this.score = score;
        var stream = getClass().getResourceAsStream(imagePath);
        if (stream == null) {
            throw new IllegalStateException("画像が見つかりません: " + imagePath);
        }
        //this.image = new Image(stream);
        
        this.image = new Image(
            stream,
            80,    // 幅50px
            0,     // 高さは比率維持で自動計算
            true,  // 比率維持
            true   // 高品質リサイズ
        );

    }

    private static final FruitType[] VALUES = values();

    /**
     * 全種類の中からランダムに1つ選んで返す。
     * MapData#spawnFruit() から呼ばれる。
     */
    public static FruitType random(Random random) {
        return VALUES[random.nextInt(VALUES.length)];
    }
    
    // ==================================================
 	// getter
 	// ==================================================
    public int getScore() { return score; }
    public Image getImage() { return image; }
}

File: src/main/java/Items/FruitType.js
/**
 * フルーツの種類を管理するクラス。
 * 種類ごとにスコアと画像を持つ。
 */
export class FruitType {

    // =========================
    // コンストラクタ
    // =========================
    constructor(score, imagePath) {

        // スコア
        this.score = score;

        // 画像
        this.image = new Image();

        // 読み込み失敗時
        this.image.onerror = () => {
            console.error(
                "画像が見つかりません: " +
                imagePath
            );
        };

        // 画像読み込み
        this.image.src = imagePath;
    }

    // =========================
    // getter
    // =========================

    getScore() {
        return this.score;
    }

    getImage() {
        return this.image;
    }

    // =========================
    // ランダム選択
    // =========================

    /**
     * 全種類の中からランダムに1つ返す
     * MapData.spawnFruit() から呼ばれる
     */
    static random() {

        const values = FruitType.VALUES;

        return values[
            Math.floor(
                Math.random() * values.length
            )
        ];
    }
}

// =========================
// フルーツ定義
// =========================

// サクランボ
FruitType.CHERRY =
    new FruitType(
        100,
        "../../resources/picture/sakuranbo.png"
    );

// イチゴ
FruitType.STRAWBERRY =
    new FruitType(
        300,
        "../../resources/picture/ichigo.png"
    );

// オレンジ
FruitType.ORANGE =
    new FruitType(
        500,
        "../../resources/picture/orange.png"
    );

// リンゴ
FruitType.APPLE =
    new FruitType(
        700,
        "../../resources/picture/ringo.png"
    );

// ブドウ
FruitType.GRAPE =
    new FruitType(
        1000,
        "../../resources/picture/budo.png"
    );

// enumのvalues()相当
FruitType.VALUES = [
    FruitType.CHERRY,
    FruitType.STRAWBERRY,
    FruitType.ORANGE,
    FruitType.APPLE,
    FruitType.GRAPE
];

